src/ape/utils.py

```diff
@@ -513,45 +557,106 @@ def get_all_files_in_directory(path: Path) -> List[Path]:
     Returns:
         List[pathlib.Path]: A list of files in the given directory.
     """
+
     if path.is_dir():
         return list(path.rglob("*.*"))
 
     return [path]
 
 
-class AbstractDataClassMeta(DataClassMeta, ABCMeta):
+class injected_before_use(property):
+    """
+    Injected properties are injected class variables that must be set before use
+    NOTE: do not appear in a Pydantic model's set of properties
+    """
+
+    def __get__(self, *args):
+        raise ValueError("Value not set. Please inject this property before calling.")
+
+
+class ManagerAccessBase:
+
+    # NOTE: cast is used to update the class type returned to mypy
+    account_manager: ClassVar["AccountManager"] = cast("AccountManager", injected_before_use())
+
+    chain_manager: ClassVar["ChainManager"] = cast("ChainManager", injected_before_use())
+
+    compiler_manager: ClassVar["CompilerManager"] = cast("CompilerManager", injected_before_use())
     """
-    A `data class <https://docs.python.org/3/library/dataclasses.html>`__ that
-    is also abstract (meaning it has methods that **must** be implemented in a
-    sub-class or else errors will occur). This class cannot be instantiated
-    on its own.
+    The group of compiler plugins for compiling source files. See
+    :class:`~ape.managers.compilers.CompilerManager` for more information.
+    Call method :meth:`~ape.managers.project.ProjectManager.load_contracts` in this class
+    to more easily compile sources.
     """
 
+    config_manager: ClassVar["ConfigManager"] = cast("ConfigManager", injected_before_use())
+    """
+    A reference to :class:`~ape.managers.config.ConfigManager`, which
+    manages project and plugin configurations.
+    """
```

Comment:
I think all of the docstrings should just be in `ape/__init__.py`, but not sure how the docs render for this

Reply:
I am trying to figure out if this is public or not.

The `ManagerAccessBase` is currently not showing up in the docs for some reason.

If this _is_ public, I'd be fine with avoiding repeating all the doc strs here that are the same from `__init__` and instead just say at the class level docs that this class has all the same singletons from the root ape namespace and maybe link to those docs.